### DISCUSSION TITLE
how i can use the Cuda acceleration in lammps installed by deepmd-kit offline

### DISCUSSION THREAD

**Discussion by 343333333** [deepmodeling/deepmd-kit#667] [General]
i install the deepmd-kit offline  with the `deepmd-kit-1.3.3-cuda10.0_gpu-Linux-x86_64.sh`
and then i wanna use the Cuda to accelerate the lammps, because i notice that  in the` conda list `the lammps is build with the   0_cuda10.0_gpu. dose it say that i can use Cuda？ but i when use the lammps to run my trained model ,the log file says:
```
[[mpiexec@gpu03] match_arg (../../utils/args/args.c:254): unrecognized argument sf
[mpiexec@gpu03] HYDU_parse_array (../../utils/args/args.c:269): argument matching returned error
[mpiexec@gpu03] parse_args (../../ui/mpich/utils.c:4764): error parsing input array
[mpiexec@gpu03] HYD_uii_mpx_get_parameters (../../ui/mpich/utils.c:5091): unable to parse user arguments"]
```
and the order is: `mpirun -np 20   -sf gpu -pk gpu 2 lmp < lammps.in `
so i get confused about that, or what else i should do to use Cuda acceleration

**Comment by 343333333** [comment]
By the way , i wanna ask a noob question, where i can find the `lammps/src` path in the deepmd-kit , because maybe i need to add some package in it or whatever.

**Comment by njzjz** [comment]
The LAMMPS was not built with the GPU package. We haven't tested it and cannot make sure it will work well. The current GPU support only involves DP (and TensorFlow).

The source code of building steps is open-source at https://github.com/deepmd-kit-recipes/lammps-dp-feedstock. You are free to modify it and build yourself.

The design of conda packages is to only distribute scripts and binary files, so you won't find the source code.

**Reply by 343333333** [reply]
> The LAMMPS was not built with the GPU package. We haven't tested it and cannot make sure it will work well. The current GPU support only involves DP (and TensorFlow). The source code of building steps is open-source at https://github.com/deepmd-kit-recipes/lammps-dp-feedstock. You are free to modify it and build yourself. The design of conda packages is to only distribute scripts and binary files, so you won't find the source code.
> […](#)
> On May 26, 2021, at 08:33, 343333333 ***@***.***> wrote: ﻿ By the way , i wanna ask a noob question, where i can find the lammps/src path in the conda , because maybe i need to add some package in it or whatever. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

ok, if i work it out someday, i would like to test that on gpu. thx for you answering.
